mlir/include/mlir/Analysis/DataFlowFramework.h

```diff
@@ -152,6 +152,11 @@ inline raw_ostream &operator<<(raw_ostream &os, const ProgramPoint &point) {
   return os;
 }
 
+inline raw_ostream &operator<<(raw_ostream &os, const ProgramPoint *point) {
+  point->print(os);
+  return os;
+}
+
```

Comment:
This is non-idiomatic IMO: we should instead fix the case where we log a pointer and use a `*` in front of it (this would be aligned with the other similar entities in MLIR, like `Operation *` for example).